tools/toolchains/__init__.py

```diff
@@ -849,6 +849,21 @@ def compile_command(self, source, object, includes):
 
         return None
 
+    def check_toolchain_path(self, name):
+        """Check if the path to toolchain is valid. Exit if not.
+
+        Positional arguments:
+        name -- the name of the toolchain, as used when creating compile
+                commands to execute. Valid key for TOOLCHAIN_PATHS.
+
+        No return value but causes a system exit of execution if the path
+        does not exist.
+        """
```

Comment:
It should exit gracefully with the error? Unless you need some exit value. Check the new PR.